src/codecs.ts

```diff
@@ -923,27 +923,35 @@ export type Policy = t.TypeOf<typeof Policy>;
 export type PolicyC = t.UnionC<[t.LiteralC<'redactEmail'>, t.LiteralC<'log'>]>;
 
 /** the codec of a route enabled in an AI integration */
-export type EnabledRouteC = t.TypeC<{
+export type EnabledRouteC<T extends t.Mixed> = t.TypeC<{
   /** the name of the enabled route */
-  routeName: t.Mixed;
+  routeName: T;
   /** the enabled policies */
   enabledPolicies: t.ArrayC<PolicyC>;
 }>;
 
 /** the codec of routes enabled in an AI integration */
-export type EnabledRoutesC = t.ArrayC<EnabledRouteC>;
+export type EnabledRoutesC<T extends t.Mixed> = t.ArrayC<EnabledRouteC<T>>;
 
 /** the codec of an AI Integration */
-export type AIIntegrationC = t.TypeC<{
+export type AIIntegrationC<T extends t.Mixed> = t.TypeC<{
   /** the routes enabled in the AI integration */
-  enabledRoutes: EnabledRoutesC;
+  enabledRoutes: EnabledRoutesC<T>;
 }>;
 
+/** The codec of OpenAI routeName */
+export type OpenAIRouteNameC = t.UnionC<
+  [t.LiteralC<'/v1/chat/completions'>, t.LiteralC<'/v1/embeddings'>]
+>;
+
 /**
  * The names of the OpenAI routes that we support setting policies for
  * reference: https://platform.openai.com/docs/api-reference/introduction
  */
-export const OpenAIRouteName = t.literal('/v1/chat/completions');
+export const OpenAIRouteName: OpenAIRouteNameC = t.union([
+  t.literal('/v1/chat/completions'),
+  t.literal('/v1/embeddings'),
+]);
```

Comment:
What is the difference between these?

Reply:
Between `t.literal('/v1/chat/completions')` and  `t.literal('/v1/embeddings')`?

Reply:
The short explanation is that `OpenAIRouteName` is the io-ts "value" and `OpenAIRouteNameC` is the io-ts type.

Reply:
The way I think about this, is that when you do

```
export const OpenAIRouteName: OpenAIRouteNameC = t.union([
  t.literal('/v1/chat/completions'),
  t.literal('/v1/embeddings'),
]);
```

What you get back is a "value" that can be either `v1/chat/completions` or `v1/embeddings`.

That's why if you do

```
export type OpenAIRouteName = t.TypeOf<typeof OpenAIRouteName>;
```

then you have  `type OpenAIRouteName = `v1/chat/completions`  | `v1/embeddings``

Reply:
ok flight is about to land.. will explain the way I think about `OpenAIRouteNameC` soon